### ISSUE TITLE
If dotfiles are in a git repo, very bad things happen.

### ISSUE THREAD

**Issue by tomkel** [jorgebucaran/fisher#380]
When using fisherman with GNU Stow, which is a bunch of symlinks from `~/.config` to the stow git repo, the fishfile references to the plugins get replaced. Here is an example when running `fisher up`.

```fish
tommy@lt-tkell713-mac ~/dotfiles> fisher up
OK fisherman is up to date
Updating 3 plugins
OK Latest bobthefish (macos)
OK Latest getopts (macos)
OK Latest z (macos)
Done in 1s 267ms

tommy@lt-tkell713-mac ~/dotfiles> git diff
diff --git a/fish/.config/fish/fishfile b/fish/.config/fish/fishfile
index b4c1b8e..6e0090a 100644
--- a/fish/.config/fish/fishfile
+++ b/fish/.config/fish/fishfile
@@ -1,3 +1,3 @@
-oh-my-fish/theme-bobthefish
-fisherman/getopts
-fisherman/z
+tomkel/dotfiles.git
+tomkel/dotfiles.git
+tomkel/dotfiles.git
```

Also, some unstaged/uncommitted changes in my dotfiles local git repo were erased after i ran `fisher rm plugin`. Erasing unstaged changes is a very serious issue. This may be due to the git reset at:
https://github.com/fisherman/fisherman/blob/master/fisher.fish#L736

**Comment by jbucaran** [comment]
Hi @tomkel, what do you suggest we do then?

**Comment by tomkel** [comment]
Is there a way to be explicit about which git repo you want the commands to apply to?

**Comment by sethfowler** [comment]
There are `--git-dir=<path/.git>` and `--work-tree=<path>`. Newer git can also do `git -C <path>`, which is the most minimalist option.

**Comment by bryaan** [comment]
I just ran into this issue and it seems pretty serious.  I am not using GNU stow.  My fishfile is simply symlinked from my dotfiles repo.  Whenever I run fisher up the fishfile is erased.  Is there another place this stuff is being stored?

**Comment by jorgebucaran** [comment]
@tomkel @sethfowler @bryaan I just released [fisher/releases/tag/3.0.0](https://github.com/jorgebucaran/fisher/releases/tag/3.0.0)~! 🎉  

V3 no longer uses symlinks to copy files to your fish configuration (customizable via $fisher_path) except for local packages.
